experiment.py

```diff
@@ -167,8 +185,8 @@ def retrofit_hinge_loss(self,
         """
         assert word_rep_pos_1.shape == word_rep_pos_2.shape
         assert word_rep_neg_1.shape == word_rep_neg_2.shape
-        positive_pair_distance = torch.norm(word_rep_pos_1 - word_rep_pos_2, p=2, dim=1) # TODO: need keepdim=True??
-        negative_pair_distance = torch.norm(word_rep_neg_1 - word_rep_neg_2, p=2, dim=1) # shape: (batch_size,) if keepdim=False
+        positive_pair_distance = torch.norm(word_rep_pos_1 - word_rep_pos_2, p=2, dim=-1) # NOTE(js) changed from dim=1 to dim=-1 to work with batch_size=1
```

Comment:
I don't think this is a good idea! Instead of changing the loss function here, can you change the test to add a fake batch dimension?

Reply:
I didn't have to adjust any tests, but I'll change it back.